hack/csi_client/pkg/csi/client.go

```diff
@@ -114,17 +74,11 @@ func (c *Client) NewRequest(req *Request) (err error) {
 					Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
 				},
 			},
-			VolumeAttributes: map[string]string{
-				attrIP:     req.ShareAddr,
-				attrVolume: req.ShareName,
-			},
+			VolumeAttributes: req.VolumeAttr,
 		}
 
 		if goOs == "windows" {
-			csiReq.NodePublishSecrets = map[string]string{
-				smbUser:     fmt.Sprintf("%s\\%s", req.ShareAddr, req.Username),
-				smbPassword: req.Password,
-			}
+			csiReq.NodePublishSecrets = req.Secrets
```

Comment:
Does this have to be only for Windows? Could this be useful (or at least not-harmful) for requests on Linux too?

Reply:
Secrets are ignored by anything that doesn't need them, so I guess not harmful.